src/test/java/ast/ASTNodesKtTest.kt

```diff
@@ -0,0 +1,173 @@
+package ast
+
+import antlr.TreepatVisitorImplementation
+import antlr.generate.TreepatLexer
+import antlr.generate.TreepatParser
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import org.antlr.v4.runtime.CharStreams
+import org.antlr.v4.runtime.CommonTokenStream
+import org.antlr.v4.runtime.tree.ParseTree
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+internal class ASTNodesKtTest {
+
+    private var countNode: Int = 0
+    private var countChild: Int = 0
+    private var countSibling: Int = 0
+
+    @BeforeEach
+    fun setUp() {
+        countNode = 0
+        countChild = 0
+        countSibling = 0
+    }
```

Comment:
I think si a little redundant to set the variables in the `@BeforeEach` block and in the beginning of the class. We can just set them in the  `@BeforeEach`.
```suggestion
    private lateinit var countNode: Int
    private lateinit var countChild: Int
    private lateinit var countSibling: Int

    @BeforeEach
    fun setUp() {
        countNode = 0
        countChild = 0
        countSibling = 0
    }
```